File: 20241227-NamedTuple/example1.py
#!/usr/bin/env python3

# téma: nevesített tuple (NamedTuple)

"""
      y
      ^
      |
      |
    1 -     o
      |
------+--|--|----------> x
      |  1  2
      |
"""

from typing import NamedTuple


class Point(NamedTuple):
    x: int
    y: int


def main():
    t = (2, 1)
    print(t[0])
    print("---")
    p = Point(x=2, y=1)
    print(p)


##############################################################################

if __name__ == "__main__":
    main()

File: 20241227-NamedTuple/example1.c
#include <stdio.h>

typedef struct {
    int x;
    int y;
} Point;

int main()
{
    Point p = {2, 1};

    // Point p = {
    // .y = 1,
    // .x = 2
    // };

    // Point p = {y: 1, x: 2};

    printf("Point(x=%d, y=%d)\n", p.x, p.y);

    return 0;
}
